src/libstd/sys/windows/process.rs

```diff
@@ -43,13 +43,25 @@ fn mk_key(s: &OsStr) -> OsString {
     })
 }
 
+fn ensure_no_nuls<T: AsRef<OsStr>>(str: T) -> io::Result<T> {
+    let has_nul = {
+        let bytes = str.as_ref().as_inner().inner.as_inner();
```

Comment:
Ah can this actually use the `.encode_wide()` method? That's what'll be passed down to the OS anyway, and is a little more reliably than looking at the internal bytes.